pinot-segment-local/src/main/java/org/apache/pinot/segment/local/function/FunctionEvaluatorFactory.java

```diff
@@ -96,6 +96,14 @@ public static FunctionEvaluator getExpressionEvaluator(FieldSpec fieldSpec) {
     return functionEvaluator;
   }
 
+  public static FunctionEvaluator getExpressionEvaluator(String transformExpression, boolean disableGroovy) {
```

Comment:
Let's add a method `isGroovyExpression()` instead of passing `disableGroovy` to the getter

Reply:
So are you saying that we first call `isGroovyExpression(String transformExpression, boolean disableGroovy)` and pass the return value to the getter?  i.e.

`public static FunctionEvaluator getExpressionEvaluator(String transformExpression, boolean isGroovyExpression)`.

Reply:
Basically adding a method `public static boolean isGroovyExpression(String transformExpression)`. The table config validator can call this method, and throw exception if it returns `true`

Reply:
Updated. Let me know if this was what you were thinking.